src/main/java/org/cardanofoundation/conversions/converters/SlotConversions.java

```diff
@@ -26,4 +26,26 @@ public LocalDateTime slotToTime(long absoluteSlot) {
     // for now post byron we have 1 slot = 1 second
     return genesisConfig.blockTime(EraType.Shelley, absoluteSlot);
   }
+
+  /**
+   * Computes the epoch a slot falls in.
+   *
+   * @param absoluteSlot the slot number to convert
+   * @return the Era number the slot falls in
+   */
+  public Long slotToEpoch(long absoluteSlot) {
+    if (absoluteSlot < 0L) {
+      throw new IllegalArgumentException("absoluteSlot cannot be negative");
```

Comment:
sometimes we are using IllegalArgumentException and sometimes InvalidConversionException, can we use 1 or another or these are different cases?

Reply:
gonna ditch the exception and use `IllegalArgumentException` everywhere

Reply:
gonna ditch the exception and use `IllegalArgumentException` everywhere